#coordinates of the center of the ring
int32 center_x
int32 center_y
